### PR TITLE
Use go 1.14 and go modules to build amazon-vpc-cni-plugins

### PR DESCRIPTION
**Pull request by angelcar**
### Summary

Modifies scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins to add support for go 1.14 and go modules for amazon-vpc-cni-plugins

### Implementation details

Pulling from golang:1.14 doker image and passing required golang flags to enable vendoring for go modules.

### Testing

make test
make release

New tests cover the changes: yes

### Description for the changelog

* Enhancement - Use go 1.14 and go modules to build amazon-vpc-cni-plugins

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins [hunk 1]
```diff
@@ -11,13 +11,17 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-FROM golang:1.10
+FROM golang:1.14
```
Comment (mythri-garaga) [inline scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins:14]: Can we upgrade this to 1.15(latest)?

Comment (angelcar) [reply scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins]: > Can we upgrade this to 1.15(latest)?

Yes we can =)

Comment (angelcar) [reply scripts/dockerfiles/Dockerfile.buildVPCCNIPlugins]: added 1.15 in a new commit

**Comment by angelcar** [comment]
Use go 1.15 and go modules to build amazon-vpc-cni-plugins

**Pull request merged**